test_a.py

```diff
@@ -0,0 +1,10 @@
+import a
+from a import print_a
+
+print(print_a())
+print(a.print_a())
+print(a.print_b())
+print(a.print_c())
```

Comment:
The above code works, Can anyone please review and provide feedback?

Reply:
I think the design is not correct
Consider the example
**test_a.py**
```py
import a
```
**ASR**
```py
(TranslationUnit 
    (SymbolTable
        1
        {
            a:
                (Module 
                    (SymbolTable
                        3
                        {
                            print_a:
                                (ExternalSymbol 
                                    3 
                                    print_a 
                                    5 print_a 
                                    b 
                                    [] 
                                    print_a 
                                    Public
                                ), 
                            print_b:
                                (ExternalSymbol 
                                    3 
                                    print_b 
                                    5 print_b 
                                    b 
                                    [] 
                                    print_b 
                                    Public
                                ), 
                            print_c:
                                (ExternalSymbol 
                                    3 
                                    print_c 
                                    9 print_c 
                                    c 
                                    [] 
                                    print_c 
                                    Public
                                )
                            
                        }) 
                    a 
                    [b
                    c] 
                    .false. 
                    .false.
                ), 
            b:
                (Module 
                    (SymbolTable
                        5
                        {
                            print_a:
                                (Function 
                                    (SymbolTable
                                        6
                                        {
                                            _lpython_return_variable:
                                                (Variable 
                                                    6 
                                                    _lpython_return_variable 
                                                    ReturnVar 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Character 1 -2 () []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                )
                                            
                                        }) 
                                    print_a 
                                    [] 
                                    [(= 
                                        (Var 6 _lpython_return_variable) 
                                        (StringConstant 
                                            "A" 
                                            (Character 1 1 () [])
                                        ) 
                                        ()
                                    )
                                    (Return)] 
                                    (Var 6 _lpython_return_variable) 
                                    Source 
                                    Public 
                                    Implementation 
                                    () 
                                    .false. 
                                    .false. 
                                    .false. 
                                    .false. 
                                    .false. 
                                    [] 
                                    [] 
                                    .false.
                                ), 
                            print_b:
                                (Function 
                                    (SymbolTable
                                        7
                                        {
                                            _lpython_return_variable:
                                                (Variable 
                                                    7 
                                                    _lpython_return_variable 
                                                    ReturnVar 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Character 1 -2 () []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                )
                                            
                                        }) 
                                    print_b 
                                    [] 
                                    [(= 
                                        (Var 7 _lpython_return_variable) 
                                        (StringConstant 
                                            "B" 
                                            (Character 1 1 () [])
                                        ) 
                                        ()
                                    )
                                    (Return)] 
                                    (Var 7 _lpython_return_variable) 
                                    Source 
                                    Public 
                                    Implementation 
                                    () 
                                    .false. 
                                    .false. 
                                    .false. 
                                    .false. 
                                    .false. 
                                    [] 
                                    [] 
                                    .false.
                                )
                            
                        }) 
                    b 
                    [] 
                    .false. 
                    .false.
                ), 
            c:
                (Module 
                    (SymbolTable
                        9
                        {
                            print_c:
                                (Function 
                                    (SymbolTable
                                        10
                                        {
                                            _lpython_return_variable:
                                                (Variable 
                                                    10 
                                                    _lpython_return_variable 
                                                    ReturnVar 
                                                    () 
                                                    () 
                                                    Default 
                                                    (Character 1 -2 () []) 
                                                    Source 
                                                    Public 
                                                    Required 
                                                    .false.
                                                )
                                            
                                        }) 
                                    print_c 
                                    [] 
                                    [(= 
                                        (Var 10 _lpython_return_variable) 
                                        (StringConstant 
                                            "C" 
                                            (Character 1 1 () [])
                                        ) 
                                        ()
                                    )
                                    (Return)] 
                                    (Var 10 _lpython_return_variable) 
                                    Source 
                                    Public 
                                    Implementation 
                                    () 
                                    .false. 
                                    .false. 
                                    .false. 
                                    .false. 
                                    .false. 
                                    [] 
                                    [] 
                                    .false.
                                )
                            
                        }) 
                    c 
                    [] 
                    .false. 
                    .false.
                ), 
            main_program:
                (Program 
                    (SymbolTable
                        11
                        {
                            
                        }) 
                    main_program 
                    [] 
                    []
                )
            
        }) 
    []
)
```
Instead of creating `ExternalSymbol` in the module `a`, we have to replace it with the actual Function, variables, ...
Also, we should not create the module `b` and `c` symtab.
**Reason:**
1.
```py
from a import b
```
Here `from a` will import the module `b`, so `import b` throws `b already defined` (Solution: we have to skip the `import b`)
2.
```py
import a
print(b.print_a())
```
The above code should throw an error. but it doesn't, as module `a` import module `b` also.

What are your thoughts? @certik @czgdp1807